### PR TITLE
Add the `eip update` command

### PR DESCRIPTION
**Pull request by mcorbin**
This command updates an elastic IP address.

```
./exo eip create --healthcheck-mode http --healthcheck-path "/" --healthcheck-port 8080
┼──────────┼─────────────────┼──────────────────────────────────────┼
│   ZONE   │       IP        │                  ID                  │
┼──────────┼─────────────────┼──────────────────────────────────────┼
│ ch-gva-2 │ 159.100.241.202 │ c595c5b6-8211-4497-9154-050c696eae1c │
┼──────────┼─────────────────┼──────────────────────────────────────┼


./exo eip show c595c5b6-8211-4497-9154-050c696eae1c
┼──────────────────────────┼──────────────────────────────────────┼
│     159.100.241.202      │                                      │
┼──────────────────────────┼──────────────────────────────────────┼
│ ID                       │ c595c5b6-8211-4497-9154-050c696eae1c │
│ Zone                     │ ch-gva-2                             │
│ Healthcheck Mode         │ http                                 │
│ Healthcheck Path         │ /                                    │
│ Healthcheck Port         │ 8080                                 │
│ Healthcheck Interval     │ 10                                   │
│ Healthcheck Strikes Fail │ 3                                    │
│ Healthcheck Strikes OK   │ 2                                    │
│ Healthcheck Timeout      │ 2                                    │
┼──────────────────────────┼──────────────────────────────────────┼

./exo eip update c595c5b6-8211-4497-9154-050c696eae1c --healthcheck-mode tcp --healthcheck-port 9999 --healthcheck-interval 20 --healthcheck-strikes-fail 4 --healthcheck-strikes-ok 3 --healthcheck-timeout 10
Updating the IP address "c595c5b6-8211-4497-9154-050c696eae1c" . success.
┼──────────┼─────────────────┼──────────────────────────────────────┼
│   ZONE   │       IP        │                  ID                  │
┼──────────┼─────────────────┼──────────────────────────────────────┼
│ ch-gva-2 │ 159.100.241.202 │ c595c5b6-8211-4497-9154-050c696eae1c │
┼──────────┼─────────────────┼──────────────────────────────────────┼

./exo eip show c595c5b6-8211-4497-9154-050c696eae1c
┼──────────────────────────┼──────────────────────────────────────┼
│     159.100.241.202      │                                      │
┼──────────────────────────┼──────────────────────────────────────┼
│ ID                       │ c595c5b6-8211-4497-9154-050c696eae1c │
│ Zone                     │ ch-gva-2                             │
│ Healthcheck Mode         │ tcp                                  │
│ Healthcheck Path         │                                      │
│ Healthcheck Port         │ 9999                                 │
│ Healthcheck Interval     │ 20                                   │
│ Healthcheck Strikes Fail │ 4                                    │
│ Healthcheck Strikes OK   │ 3                                    │
│ Healthcheck Timeout      │ 10                                   │
┼──────────────────────────┼──────────────────────────────────────┼

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/cenkalti/backoff v2.0.0+incompatible
 	github.com/cpuguy83/go-md2man v1.0.8 // indirect
 	github.com/dustin/go-humanize v0.0.0-20180713052910-9f541cc9db5d
-	github.com/exoscale/egoscale v0.15.0
+	github.com/exoscale/egoscale v0.15.1-0.20190405090039-2bfdebb29af9
```
Comment (falzm) [inline go.mod]: Would be nice if you could release a tagged version 🙏

Comment (pierre-emmanuelJ) [reply go.mod]: 👍

File: cmd/eip_update.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+func init() {
+	eipUpdateCmd.Flags().Int64P("healthcheck-interval", "", 0, "the time in seconds to wait for between each healthcheck.")
+	eipUpdateCmd.Flags().StringP("healthcheck-mode", "", "", "the healthcheck type. Should be tcp or http.")
+	eipUpdateCmd.Flags().StringP("healthcheck-path", "", "", "the healthcheck path. Required if mode is http.")
+	eipUpdateCmd.Flags().Int64P("healthcheck-port", "", 0, "the healthcheck port (e.g. 80 for http).")
+	eipUpdateCmd.Flags().Int64P("healthcheck-strikes-fail", "", 0, "the number of times to retry before declaring the IP dead.")
+	eipUpdateCmd.Flags().Int64P("healthcheck-strikes-ok", "", 0, "the number of times to retry before declaring the IP healthy.")
+	eipUpdateCmd.Flags().Int64P("healthcheck-timeout", "", 0, "the timeout in seconds to wait for each check (default is 2). Should be lower than the interval.")
```
Comment (pierre-emmanuelJ) [inline cmd/eip_update.go]: possible to find some short flags for them?

Comment (falzm) [reply cmd/eip_update.go]: > possible to find some short flags for them?

IMO it's better to keep them explicit.

**Comment by falzm** [comment]
> LGTM Waiting for egoscale release

@pierre-emmanuelJ We're waiting on you for this ;) https://github.com/exoscale/egoscale/pull/397

File: cmd/eip_update.go (hunk 2 above)
Comment (pierre-emmanuelJ) [reply cmd/eip_update.go]: Agree but I speak about to add short flag and for sure keep long flags e.g:
```Golang
eipUpdateCmd.Flags().Int64P("healthcheck-port", "p", 0, "the healthcheck port (e.g. 80 for http).")
```
`p`

Comment (mcorbin) [reply cmd/eip_update.go]: For commands with a lot of flags, I don't know what short flag to use (for example for healthcheck-strikes-ok, healthcheck-strikes-timeout...).

**Pull request merged**